test/extended/router/gatewayapicontroller.go

```diff
@@ -246,6 +249,90 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 		g.By("Validating the http connectivity to the backend application")
 		assertHttpRouteConnection(defaultRoutename)
 	})
+
+	g.It("Ensure OSSM subscription, istiod deployment and the istio could be deleted and then get recreated [Serial]", func() {
+		const (
+			operatorNamespace      = "openshift-operators"
+			ingressNamespace       = "openshift-ingress"
+			ossmSubscriptionName   = "servicemeshoperator3"
+			csvName                = "servicemeshoperator3.v3.0.0"
+			ossmOperatorDeployment = "servicemesh-operator3"
+			istiodDeployment       = "istiod-openshift-gateway"
+			istioName              = "openshift-gateway"
+		)
+
+		// deleted the OSSM subscription and then checked if it was restored
+		g.By(fmt.Sprintf("Try to delete the subscription %s", ossmSubscriptionName))
+		_, err := oc.AsAdmin().WithoutNamespace().Run("delete").Args("-n", operatorNamespace, "subscription/"+ossmSubscriptionName).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By(fmt.Sprintf("Wait until the the OSSM subscription %s is automatically created successfully", ossmSubscriptionName))
+		pollWaitSubscriptionCreated(oc, operatorNamespace, ossmSubscriptionName)
+
+		g.By(fmt.Sprintf("Wait until the the OSSM csv %s is automatically created successfully", csvName))
+		pollWaitOssmCsvCreated(oc, operatorNamespace, csvName)
+
+		// deleted the istiod deployment and then checked if it was restored
+		deleteDeploymentAndWaitAvailableAgain(oc, istiodDeployment, ingressNamespace)
+
+		// deleted the istio and check if it was restored
+		g.By(fmt.Sprintf("Try to delete the istio %s", istioName))
+		output, err := oc.AsAdmin().WithoutNamespace().Run("delete").Args("-n", ingressNamespace, "istio/"+istioName).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(output).To(o.ContainSubstring("deleted"))
```

Comment:
Why should we do this additional check?

Reply:
It is a negative test, I think. @lihongan Can we remove this test(or we must keep it)?

Reply:
I think @alebedev87 means the check `o.Expect(output).To(o.ContainSubstring("deleted"))` is not required, since `o.Expect(err).NotTo(o.HaveOccurred())` is enough. right ?